src/hub_predtimechart/generate_options.py

```diff
@@ -3,9 +3,33 @@
 
 def ptc_options_for_hub(hub_config: HubConfig):
     """
-    xx
+    Returns predtimechart options dict for `hub_config` - see https://github.com/reichlab/predtimechart?tab=readme-ov-file#options-object .
 
     :param hub_config: a HubConfig
-    :return: xx
     """
-    return {}  # todo xx
+    options = {}
+
+    # todo xx finish!
+
+    # options['target_variables'] = None
+    # options['initial_target_var'] = None
+    # options['task_ids'] = None
+    # options['initial_task_ids'] = None
+
+    options['intervals'] = ["0%", "50%", "95%"]
+    options['initial_interval'] = options['intervals'][-1]
+
+    # options['available_as_ofs'] = hub_config.fetch_reference_dates
+    # options['initial_as_of'] = None
+    # options['current_date'] = None
+
+    options['models'] = []
+    for model_id, metadata in hub_config.model_id_to_metadata.items():
+        if metadata['designated_model']:  # todo xx put here?: predtimechart-config.yml
```

Comment:
Is `designated_model` a status that we'd expect to change from round to round? (thanks for defining that phrase so well in the README)?

Reply:
Jeesh - I don't know. But if we assume that the config and json files will be (re)generated at each round, any changes should be picked up, right?

Reply:
Ok, trying to think this through because it's not clear how this is supposed to work.

If a team updates its `designated model` value halfway through the hub's operation, should the viz retroactively apply the new value to prior rounds? Or is the viz supposed to reflect the model metadata state as it was when prior rounds closed (i.e., point in time)?

(AFAIK, model metadata files are submitted by teams as a prerequisite to participating in the hub, and they remain fairly static. For example, here are the [files from last year's FluSight hub](https://github.com/cdcepi/FluSight-forecast-hub/tree/main/model-metadata))

Reply:
OK, I see what you're getting at. Maybe we need an UpdateVizConfig as well as UpdateVizData. Thanks.s